plugins/modules/gitlab_project.py

```diff
@@ -471,7 +517,13 @@ def update_project(self, project, arguments):
         for arg_key, arg_value in arguments.items():
             if arguments[arg_key] is not None:
                 if getattr(project, arg_key) != arguments[arg_key]:
-                    setattr(project, arg_key, arguments[arg_key])
+                    if arg_key == 'container_expiration_policy':
+                        old_val = getattr(project, arg_key)
+                        if all(old_val.get(key) == value for key, value in arg_value.items()):
```

Comment:
Shouldn't this be:
```suggestion
                        if all(old_val.get(key) == value for key, value in arg_value.items() if value is not None):
```
?